RosNet/DataModel/Time.cs

```diff
@@ -4,7 +4,7 @@
 
 namespace RosNet.DataModel
 {
-    public class Time
+    public class Time : IComparable, ITime
```

Comment:
```suggestion
    public class Time : IComparable<Time>, ITime
```
https://docs.microsoft.com/en-us/dotnet/api/system.icomparable-1?view=net-6.0, and using a `record` implements `IEquatable<self>` automatically (though a little unrelated, this class sohuld be a record due to  immutability) https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record